### PR TITLE
task: Subsume keys to check

### PR DESCRIPTION
**Pull request by chriswk**
This collapses the keys seen. Removing keys that have been subsumed by a wider key (a key that includes the same projects or more as existing keys).

### CODE REVIEW

**Comment by chriswk** [comment]
One thing. Quite a bit of this logic seems like it's the same no matter where we store things, so if anyone wants to have a once-over on our abstraction layer I think that could simplify adding new datasources.

File: server/src/data_sources/memory_provider.rs [hunk 1]
```diff
@@ -51,11 +51,22 @@ impl MemoryProvider {
...
         }
+        self.reduce_tokens_to_refresh();
+    }
+    fn reduce_tokens_to_refresh(&mut self) {
+        let tokens: Vec<EdgeToken> = self
+            .tokens_to_refresh
+            .values()
+            .map(|r| r.token.clone())
+            .collect();
+        let minimized = crate::tokens::simplify(&tokens);
```
Comment (gastonfournier) [inline server/src/data_sources/memory_provider.rs:66]: I think it's time to remove:
https://github.com/Unleash/unleash-edge/blob/ca0a50d711f8c504f2ad9671929abc663639264b/server/src/tokens.rs#L3

**Review by gastonfournier — APPROVED** [review]
Looks good!

**Review by sighphyre — APPROVED** [review]
LG

**Pull request merged**